src/main/java/it/gov/pagopa/pu/workflow/wf/receiptclassification/iuf/classification/IufReceiptClassificationWFImpl.java

```diff
@@ -0,0 +1,122 @@
+  private TransferClassificationActivity transferClassificationActivity;
+
+  private TransferClassificationWfHelperActivity transferClassificationWfHelperActivity;
+
+  private IufClassificationActivityResult result;
+
+  /**
+   * Temporal workflow will not allow to use injection in order to avoid
+   * <a href="https://docs.temporal.io/workflows#non-deterministic-change">non-deterministic changes</a> due to dynamic reconfiguration.<BR />
+   * Anyway it allows to override ActivityOptions, but actually it's not supporting the override based on the particular workflow.<BR />
+   * In {@link it.gov.pagopa.pu.workflow.config.TemporalWFImplementationCustomizer} we are already setting defaults to all workflows.<BR />
+   * Use this as an example to override based on the particular workflow.
+   */
+  @Override
+  public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+    IufReceiptClassificationWfConfig wfConfig = applicationContext.getBean(IufReceiptClassificationWfConfig.class);
+
+    clearClassifyIufActivity = wfConfig.buildClearClassifyIufActivityStub();
+    iufClassificationActivity = wfConfig.buildIufClassificationActivityStub();
+    transferClassificationActivity = wfConfig.buildTransferClassificationActivityStub();
+
+    transferClassificationWfHelperActivity = wfConfig.buildTransferClassificationStarterHelperActivityStub();
+
+  }
+
+  @Override
+  public void classify() {
+
+    if (result != null && result.isSuccess()) {
+
+      // signal transfer classification for each transfer
+      result.getTransfers2classify().forEach(transfer2ClassifyDTO -> {
+
+        String iuv = transfer2ClassifyDTO.getIuv();
+        String iur = transfer2ClassifyDTO.getIur();
+        int transferIndex = transfer2ClassifyDTO.getTransferIndex();
+
+        transferClassificationWfHelperActivity.signalTransferClassificationWithStart(result.getOrganizationId(), iuv, iur, transferIndex);
+      });
+
+    } else {
+      log.warn("Result is null or classification was not successful for organizationId {}",
+        result != null ? result.getOrganizationId() : "N/A");
+    }
+
+  }
+
+
+  @Override
+  public void signalForTreasury(IufReceiptClassificationForTreasurySignalDTO signalDTO) {
+
+    log.info("Handling iuf receipt classification for treasury with organizationId {}, treasuryId {} and IUF {}",
+      signalDTO.getOrganizationId(), signalDTO.getTreasuryId(), signalDTO.getIuf());
+
+    boolean clearedForTreasury = clearClassifyIufActivity.deleteClassificationByIuf(
+      signalDTO.getOrganizationId(),
+      signalDTO.getIuf());
+
+    log.info("IUF receipt classification cleared with result {} for organizationId {}, treasuryId {} and IUF {}",
+      clearedForTreasury,
+      signalDTO.getOrganizationId(), signalDTO.getTreasuryId(), signalDTO.getIuf());
+
+    result = iufClassificationActivity.classify(signalDTO.getOrganizationId(), signalDTO.getTreasuryId(), signalDTO.getIuf());
```

Comment:
the result should not be override each time, otherwise, if multiple signals are handled, just the last would store its result!

result should be a list instantiated on the class itself, here we should just add new results